### ISSUE TITLE
Dart 1.13.0 breaks build

### ISSUE THREAD

**Issue by bunjiboys** [Netflix/security_monkey#259]
The most recent release of Dart (v1.13.0) breaks the build for the webapp. Workaround for now is to install dart 1.12.x instead. For example for Ubuntu `apt-get install dart=1.12.2-1`

Output from `pub get`
```
root@localhost:/opt/security_monkey# /usr/lib/dart/bin/pub get
Resolving dependencies...
Got dependencies!
Precompiling executables...
Loading source assets...
Loading di/module_transformer and smoke/src/default_transformer transformers...
Precompiled angular:parser_generator_for_spec.
Precompiled angular:pub_build.
Precompiled angular:template_cache_generator.
Precompiled angular:expression_extractor.
```
Output from `pub build`
```
root@localhost:/opt/security_monkey# /usr/lib/dart/bin/pub build
Loading source assets...
Loading dart_to_js_script_rewriter, di/module_transformer, initialize/build/loader_replacer, smoke/src/default_transformer and initialize transformers... (1.4s)
Loading angular transformers...
Building security_monkey... (1.2s)
[Error from InjectorGenerator on security_monkey|web/main.dart]:
Unable to resolve asset ID for "dart:core"
Build error:
Transform InjectorGenerator on security_monkey|web/main.dart threw error: <rethrow>
Caused by Could not resolve dart:core
#0      AnalysisTask._safelyPerform (package:analyzer/src/generated/engine.dart:6762:7)
#1      AnalysisTask.perform (package:analyzer/src/generated/engine.dart:6727:7)
#2      AnalysisContextImpl._cacheDartResolutionData (package:analyzer/src/generated/engine.dart:2414:79)
#3      AnalysisContextImpl._getDartResolutionData (package:analyzer/src/generated/engine.dart:3046:17)
#4      AnalysisContextImpl._getDartResolutionData2 (package:analyzer/src/generated/engine.dart:3077:14)
#5      AnalysisContextImpl.computeLibraryElement (package:analyzer/src/generated/engine.dart:1192:58)
#6      ResolverImpl._performResolve.<anonymous closure>.<anonymous closure> (package:code_transformers/src/resolver_impl.dart:158:25)
#7      MappedListIterable.elementAt (dart:_internal/iterable.dart:413)
#8      ListIterable.toList (dart:_internal/iterable.dart:219)
#9      ResolverImpl._performResolve.<anonymous closure> (package:code_transformers/src/resolver_impl.dart:159:10)
#10     _RootZone.runUnary (dart:async/zone.dart:1149)
#11     _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:502)
#12     _Future._propagateToListeners (dart:async/future_impl.dart:585)
#13     _Future._completeWithValue (dart:async/future_impl.dart:376)
#14     _Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:430)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:43)
#16     _microtaskLoopEntry (dart:async/schedule_microtask.dart:52)
#17     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#18     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)


package:analyzer/src/generated/engine.dart 2115:11       AnalysisContextImpl.recordResolveDartLibraryTaskResults
package:analyzer/src/generated/engine.dart 5304:98       AnalysisContextImpl_AnalysisTaskResultRecorder.visitResolveDartLibraryTask
package:analyzer/src/generated/engine.dart 14327:50      ResolveDartLibraryTask.accept
package:analyzer/src/generated/engine.dart 6732:12       AnalysisTask.perform
package:analyzer/src/generated/engine.dart 2414:79       AnalysisContextImpl._cacheDartResolutionData
package:analyzer/src/generated/engine.dart 3046:17       AnalysisContextImpl._getDartResolutionData
package:analyzer/src/generated/engine.dart 3077:14       AnalysisContextImpl._getDartResolutionData2
package:analyzer/src/generated/engine.dart 1192:58       AnalysisContextImpl.computeLibraryElement
package:code_transformers/src/resolver_impl.dart 158:25  ResolverImpl._performResolve.<fn>.<fn>
dart:_internal/iterable.dart 413                         MappedListIterable.elementAt
dart:_internal/iterable.dart 219                         ListIterable.toList
package:code_transformers/src/resolver_impl.dart 159:10  ResolverImpl._performResolve.<fn>
dart:async/zone.dart 1149                                _RootZone.runUnary
dart:async/future_impl.dart 502                          _Future._propagateToListeners.handleValueCallback
dart:async/future_impl.dart 585                          _Future._propagateToListeners
dart:async/future_impl.dart 376                          _Future._completeWithValue
dart:async/future_impl.dart 430                          _Future._asyncComplete.<fn>
dart:async/schedule_microtask.dart 43                    _microtaskLoop
dart:async/schedule_microtask.dart 52                    _microtaskLoopEntry
dart:isolate-patch/isolate_patch.dart 96                 _runPendingImmediateCallback
dart:isolate-patch/isolate_patch.dart 151                _RawReceivePortImpl._handleMessage

package:analyzer/src/generated/engine.dart 2115:11       AnalysisContextImpl.recordResolveDartLibraryTaskResults
package:analyzer/src/generated/engine.dart 5304:98       AnalysisContextImpl_AnalysisTaskResultRecorder.visitResolveDartLibraryTask
package:analyzer/src/generated/engine.dart 14327:50      ResolveDartLibraryTask.accept
package:analyzer/src/generated/engine.dart 6732:12       AnalysisTask.perform
package:analyzer/src/generated/engine.dart 2414:79       AnalysisContextImpl._cacheDartResolutionData
package:analyzer/src/generated/engine.dart 3046:17       AnalysisContextImpl._getDartResolutionData
package:analyzer/src/generated/engine.dart 3077:14       AnalysisContextImpl._getDartResolutionData2
package:analyzer/src/generated/engine.dart 1192:58       AnalysisContextImpl.computeLibraryElement
package:code_transformers/src/resolver_impl.dart 158:25  ResolverImpl._performResolve.<fn>.<fn>
dart:_internal                                           ListIterable.toList
package:code_transformers/src/resolver_impl.dart 159:10  ResolverImpl._performResolve.<fn>

Build failed.
```

**Comment by monkeysecurity** [comment]
I'll update the docs to pin to the old version.

**Comment by monkeysecurity** [comment]
PR #263 to update docs and the install script to pin to v 1.12.2-1.

**Comment by monkeysecurity** [comment]
merged

**Comment by monkeysecurity** [comment]
@jinxicmu left a comment on https://github.com/dart-lang/sdk/issues/22099 indicating a fix could be to pin to `"angular: "^1.1.2+2"` and let dart update.

Opening #357 to update pubspec.yaml and update the docs to again allow the latest dart version.
